Component/date item

abdo-essam opened the pull request:
![image](https://github.com/user-attachments/assets/3f2f7db8-2da6-4dd1-8b08-c965b960ca36)
Make sure that you add both states for selected and default for the date item.
Don't forget to gradient the color from start to end.

abdo-essam committed:
feat: Implement CategoryItem component

This commit introduces the `CategoryItem` composable, a reusable UI
component for displaying category information. It includes an icon,
title, and optional count and selection indicator.  It also adds
the necessary vector drawables for the category icons.

abdo-essam committed:
refactor: Reduce size of category item title

Updated the typography style for the category item title from
`labelMedium` to `labelSmall` to better fit within the item's
layout and improve visual balance.

abdo-essam committed:
feat: Implement DateItem composable

This commit introduces the DateItem composable, a UI component
that displays a date and day of the week. It includes styling
for selected and unselected states, with different background
gradients and text colors. The component is also clickable.
A preview is included for easy testing and visualization.

abdo-essam committed:
fix: Adjust check icon color in CategoryItem

The check icon color in the CategoryItem was previously hardcoded
to white. This change updates the color to use the `onPrimary`
color from the `TudeeTheme` with an alpha of 0.87 for better
contrast and consistency with the application's theme.

abdo-essam committed:
feat: Implement custom buttons and text field

Adds new custom button components (Primary, Secondary, Negative, Text)
and a custom text field for consistent UI styling and functionality.
Also includes a loading animation and new icons. Removes the old
TudeeButton. Updates colors and adds Arabic string resources.

abdo-essam committed:
refactor: Use TudeeTheme colors for DateItem gradient

Replaces hardcoded colors in the DateItem composable's
gradient with colors from the TudeeTheme. This promotes
consistency and maintainability across the application's
UI.

abdo-essam commented on app/src/main/java/com/london/tudee/presentation/components/DateItem.kt:
```diff
@@ -0,0 +1,97 @@
+                        colors = listOf(TudeeTheme.colors.surfaceLow, TudeeTheme.colors.surfaceLow)
+                    )
+                },
+                shape = RoundedCornerShape(16.dp)
```
done

abdo-essam committed:
refactor: Use TudeeTheme shapes and spacing in DateItem

Use predefined theme values for shape and padding in DateItem
for consistency.

abdo-essam committed:
feat: Add date item components and assets

This commit introduces new UI components: SnackBar and TaskStatusSlider,
along with supporting drawables (emojis, images) for date item
display and feedback. It also updates the commit-message hook.
The Task and TaskDao Entities were deleted, indicating an architecture refactor.

BassantM commented on app/src/main/java/com/london/tudee/presentation/components/DateItem.kt:
```diff
@@ -0,0 +1,89 @@
+        )
+
+        Text(
+            text = dayOfWeek,
```
style will be labelMedium instead of labelSmall

abdo-essam committed:
feat: Improve DateItem text visibility

Updated DateItem's day of week text style to `labelMedium`
to improve readability. Also, removed unused `app_name` string
resource.

abdo-essam committed:
feat: Refactor DateItem and theme colors

Updates DateItem background and adjusts light and dark theme colors
for improved visual consistency and aesthetics.

abdo-essam committed:
refactor: Adjust color values for light and dark themes

Refactors color definitions in LightThemeColors.kt and
DarkThemeColors.kt to improve visual consistency and contrast.
This includes changes to surface colors, text colors, and
various accent colors for better readability and aesthetics.

abdo-essam committed:
refactor: Rename `day` to `dayOfMonth` in DateItem

This change renames the `day` property in the `DateItem`
composable to `dayOfMonth` for clarity and consistency.
